### PR TITLE
Add typescript support

### PR DESCRIPTION
**Pull request by evanshortiss**
This is an example of addressing #2 if desired in this repository.

### CODE REVIEW

File: fastify-url-data.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  path: string
+  host: string
+  port: number
+  query: string
+}
+
+// Extend FastifyReply with the "fastify-url-data" function
+declare module 'fastify' {
+  interface FastifyRequest {
+    urlData (): UrlObject
```
Comment (evanshortiss) [inline fastify-url-data.d.ts]: This should use `urijs` types. I noticed after pushing, can fix post discussion in Issues 😄

**Comment by wong2** [comment]
@evanshortiss 👍 didn't know typescript interfaces can be extended like this

**Comment by evanshortiss** [comment]
@wong2 yep, it's pretty neat - definitely not perfect since it does not handle scoping and assumes any instance of `FastifyRequest` in that file will have the `urlData` method which might not be the case depending on how the application is configured, but 9 of 10 times it's fine.

**Comment by jsumners** [comment]
I am very 👎 on this. I do not see a reason to maintain TS stuff in such a simple plugin. It adds a lot of complexity for very little benefit. Plus, any maintenance I do to the plugin will _not_ see updates to the TS things because I do not work with TS.

**Comment by evanshortiss** [comment]
@mcollina I think having something documented is a great idea. Should it be in the core `docs` under it's own file? It'd allow people to contribute back much more easily 👍

**Comment by jsumners** [comment]
The can this PR be closed?

**Comment by evanshortiss** [comment]
I think it can be closed.

I still think shipping types with the module is better for the community and users as recommended by DefinitelyTyped.

We'll probably need a standard on that front though. Should be easy enough to document that all plugins are available via DefinitelyTyped, i.e `npm i @types/fastify-${PLUGIN_NAME}` while core types are bundled with fastify. I'm doing a docs PR now

**Pull request closed without merging**